File: backend/app/models/usuario.py
import enum
from datetime import datetime

from sqlalchemy import DateTime, Enum, String, Boolean, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class RolUsuario(str, enum.Enum):
    admin = "admin"
    granjero = "granjero"


class Usuario(Base):
    __tablename__ = "usuarios"

    id: Mapped[int] = mapped_column(primary_key=True)
    nombre: Mapped[str] = mapped_column(String(120))
    email: Mapped[str] = mapped_column(String(255), unique=True, index=True)
    password_hash: Mapped[str] = mapped_column(String(255))
    rol: Mapped[RolUsuario] = mapped_column(Enum(RolUsuario))
    activo: Mapped[bool] = mapped_column(Boolean, default=True)
    creado_en: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: mobile/lib/core/usuario.dart
enum RolUsuario { admin, granjero }

RolUsuario rolDesdeString(String valor) {
  return valor == 'admin' ? RolUsuario.admin : RolUsuario.granjero;
}

/// Espejo de `UsuarioOut` del backend (app/schemas/usuario.py).
class Usuario {
  Usuario({required this.id, required this.nombre, required this.email, required this.rol});

  final int id;
  final String nombre;
  final String email;
  final RolUsuario rol;

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'] as int,
      nombre: json['nombre'] as String,
      email: json['email'] as String,
      rol: rolDesdeString(json['rol'] as String),
    );
  }
}
